File: C-library/examples/Fibonacci/Fibonacci.r
#! /usr/bin/env rebol
REBOL [
	Title:		"Fibonacci numbers"
	Author:		"Kaj de Vos"
	Rights:		"Copyright (c) 2012 Kaj de Vos"
	License: {
		PD/CC0
		http://creativecommons.org/publicdomain/zero/1.0/
	}
;	Needs:		"REBOL 3 or 2"
	Tabs:		4
]

parameter: 40

fibonacci: func [n [integer!]] [
	either n < 2 [n] [(fibonacci n - 1) + fibonacci n - 2]
]

start: now/time/precise

prin ["Fibonacci" parameter] print [":" fibonacci parameter]
print ["Elapsed time:" now/time/precise - start]

File: C-library/examples/Fibonacci/Fibonacci.c
/*
	Fibonacci numbers
	Copyright (c) 2012 Kaj de Vos
	License: PD/CC0
		http://creativecommons.org/publicdomain/zero/1.0/
*/

#include <stdio.h>
#include <time.h>

#define parameter 40

int fibonacci (int n) {
	return n < 2 ? n : fibonacci (n - 1) + fibonacci (n - 2);
}

int main (int argc, const char * argv []) {
	time_t start = time (NULL);
	int f = fibonacci (parameter);

	printf ("Fibonacci %i: %i\nElapsed time: %f\nProcess time: %f\n",
		parameter, f,
		difftime (time (NULL), start),
		(double) clock () / CLOCKS_PER_SEC
	);
}
